src/algebra/ring/aut.lean

```diff
@@ -60,4 +64,38 @@ by refine_struct { to_fun := ring_equiv.to_mul_equiv }; intros; refl
 def to_perm : ring_aut R →* equiv.perm R :=
 by refine_struct { to_fun := ring_equiv.to_equiv }; intros; refl
 
+end mul_add
+
+section semiring
+
+variables (R G : Type*) [group G] [semiring R]
+
+/-- The tautological action by the group of automorphism of a ring `R` on `R`.-/
+instance apply_mul_semiring_action :
+mul_semiring_action (ring_aut R) R :=
+{ smul := ($),
+  smul_zero := ring_equiv.map_zero,
+  smul_add := ring_equiv.map_add,
+  smul_one := ring_equiv.map_one,
+  smul_mul := ring_equiv.map_mul,
+  one_smul := λ _, rfl,
+  mul_smul := λ _ _ _, rfl }
+
+@[simp]
+protected lemma
+smul_def (f : ring_aut R) (r : R) : f • r = f r := rfl
+
+instance apply_has_faithful_smul : has_faithful_smul (ring_aut R) R := ⟨λ _ _, ring_equiv.ext⟩
+
+/-- Each element of the group defines a ring automorphism.
+
+This is a stronger version of `distrib_mul_action.to_add_aut` and
+ `mul_distrib_mul_action.to_mul_aut`. -/
+@[simps] def mul_semiring_action.to_ring_aut [mul_semiring_action G R] : G →* ring_aut R :=
```

Comment:
This creates a definition called `ring_aut.mul_semiring_action.to_ring_aut`, which probably isn't what  you want; instead
```suggestion
@[simps] def _root_.mul_semiring_action.to_ring_aut [mul_semiring_action G R] : G →* ring_aut R :=
```